proposals/20220203-versioning-schema-amendment.md

```diff
@@ -0,0 +1,61 @@
+
+This amendment introduces two different version numbers for releasing artifacts (instead of one single version number):
+
+- The **libs version number** which represents the build version of the user-space libraries (i.e., libscap, libsinsp, and possibly any other further user-space library)
+- The **driver version number** which represents the build version of kernel-space drivers (i.e., the kernel module, the eBPF probe, and possibly any other kernel-space driver)
+
+This proposal does not aim to introduce changes other than using two different version numbers. Moreover, this proposal is only about the versions string used at build time, so no code changes are expected outside the build system context.
+
+## Motivation
+
+The [20210818-driver-semver.md](20210818-driver-semver.md#motivation) proposal is already fully implemented. The advantages introduced by that proposal would be lost if we used a single versioning number for all the artifacts as mandated by [20210524-versioning-and-release-of-the-libs-artifacts.md#versioning-scheme](20210524-versioning-and-release-of-the-libs-artifacts.md#versioning-scheme). Furthermore, that proposal is only about versioning the APIs that sits between the kernel and the user-space components. It does not mandate any specification regarding the versioning of artifacts.
+
+This amendment is required to fill the gap between the two proposals and allow our users to reuse the same driver version across a range of compatible consumers.
+
+## Goals
+
+* Document the two versioning schemes (one for the libs and another for the drivers)
+* Allow releasing libs and drivers separately (different timing and versioning)
+* Make the drivers reusable across libscap consumers and their versions
+
+## Non-goals
+
+* Introduce code changes other than in the build system and in the documentation
+* Indicate how the distribution of the artifacts must be implemented
+
+## Proposed solution
+
+### Userspace libs artifacts
+
+Libscap and libsinsp are two distinct artifacts. They will be released with the same version number (a single [SemVer 2.0](https://semver.org/spec/v2.0.0.html) string). This proposal does not aims to change what already proposed by [20210524-versioning-and-release-of-the-libs-artifacts.md#versioning-scheme](20210524-versioning-and-release-of-the-libs-artifacts.md#versioning-scheme) with respect to the user-space components residing in `falcosecurity/libs`.
+
+
+### Drivers artifacts
+
+The kernel model and the eBPF probe are two components that can be built for any supported Kernel version. The source code of both drivers will be released with the same version number (a single [SemVer 2.0](https://semver.org/spec/v2.0.0.html) string).
```

Comment:
Typo: `module` i guess!

Reply:
good catch thank you!

Reply:
Fixed by https://github.com/falcosecurity/libs/pull/198/commits/c9b207f5f3521e63e6cf52665b32a6177a624a7c